File: src/engines/numerology/services/elementMapper.py
"""
Service to translate a raw numerology number into an elemental vector.
"""
from ..constants import parameters

def number_to_vector(number):
    """
    Look up the element vector for a given number.
    Returns a copy of the dictionary to prevent mutation issues.
    """
    if number in parameters.ELEMENT_VECTORS:
        return parameters.ELEMENT_VECTORS[number].copy()
    
    # Fallback (Should ideally not happen if reduction works correctly)
    return {"air": 0, "water": 0, "fire": 0, "earth": 0, "spirit": 0}

File: src/server/engines/numerology/elementMapper.ts
import type { ElementVector } from "@/types/personalization";
import { ELEMENT_VECTORS } from "./constants";

const ZERO_VECTOR: ElementVector = {
  air: 0,
  water: 0,
  fire: 0,
  earth: 0,
  spirit: 0,
};

/** Map a numerology number to its element vector. Returns a copy. */
export function numberToVector(n: number): ElementVector {
  const vec = ELEMENT_VECTORS[n];
  if (vec === undefined) {
    return { ...ZERO_VECTOR };
  }
  return { ...vec };
}
